pkg/handler/repositories.go

```diff
@@ -318,3 +326,57 @@ func (rh *RepositoryHandler) deleteRepository(c echo.Context) error {
 	}
 	return c.NoContent(http.StatusNoContent)
 }
+
+// IntrospectRepository godoc
+// @summary 		introspect a repository
+// @ID				introspect
+// @Tags			repositories
+// @Param  			uuid            path    string  true  "Identifier of the Repository"
+// @Param			reset_count     query   bool    false "Reset the failed introspections count"
+// @Success			204 "Introspection was successfully queued"
+// @Failure      	400 {object} ce.ErrorResponse
+// @Failure      	404 {object} ce.ErrorResponse
+// @Failure      	500 {object} ce.ErrorResponse
+// @Router			/repositories/{uuid} [post]
+func (rh *RepositoryHandler) introspect(c echo.Context) error {
+	var msg *message.IntrospectRequestMessage
+	var resetCount bool
+
+	_, orgID := getAccountIdOrgId(c)
+	uuid := c.Param("uuid")
+
+	echo.QueryParamsBinder(c).Bool("reset_count", &resetCount)
+
+	response, err := rh.RepositoryConfigDao.Fetch(orgID, uuid)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository", err.Error())
+	}
+
+	repo, err := rh.RepositoryDao.FetchForUrl(response.URL)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository uuid", err.Error())
+	}
+
+	limit := time.Second * time.Duration(config.Get().Options.IntrospectApiTimeLimitSec)
+	since := time.Since(*repo.LastIntrospectionTime)
+	if since < limit {
+		detail := fmt.Sprintf("This repository has been introspected recently. Try again in %v", (limit - since).Truncate(time.Second))
+		return ce.NewErrorResponse(http.StatusBadRequest, "Error introspecting repository", detail)
```

Comment:
I'm treating this time limit like an error, but maybe we just want it to quietly skip introspection?

Reply:
I think a 400 is fine here.  We'd probably want to prevent it in the UI, so a normal user would never see it.